victims/victims_test.go

```diff
@@ -0,0 +1,223 @@
+package victims
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/asobti/kube-monkey/config"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kube "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	NAMESPACE  = metav1.NamespaceDefault
+	IDENTIFIER = "kube-monkey-id"
+)
+
+func newPod(name string, status v1.PodPhase) v1.Pod {
+
+	return v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: NAMESPACE,
+			Labels: map[string]string{
+				"kube-monkey/identifier": IDENTIFIER,
+			},
+		},
+		Status: v1.PodStatus{
+			Phase: status,
+		},
+	}
+
+}
+
+func newVictimBase() *VictimBase {
+	return New("Pod", "name", NAMESPACE, IDENTIFIER, 1)
+}
+
+func getPodList(client kube.Interface) *v1.PodList {
+	podList, _ := client.CoreV1().Pods(NAMESPACE).List(metav1.ListOptions{})
+	return podList
+}
+
+func TestVictimBaseTemplateGetters(t *testing.T) {
+
+	v := newVictimBase()
+
+	assert.Equal(t, "Pod", v.Kind())
+	assert.Equal(t, "name", v.Name())
+	assert.Equal(t, NAMESPACE, v.Namespace())
+	assert.Equal(t, IDENTIFIER, v.Identifier())
+	assert.Equal(t, 1, v.Mtbf())
+}
+
+func TestRunningPods(t *testing.T) {
+
+	v := newVictimBase()
+	pod1 := newPod("app1", v1.PodRunning)
+	pod2 := newPod("app2", v1.PodPending)
+
+	client := fake.NewSimpleClientset(&pod1, &pod2)
+
+	podList, e := v.RunningPods(client)
+
+	if e != nil {
+		t.Errorf("Unexpected error %s while getting running pod", e)
+	}
+
+	if len(podList) != 1 {
+		t.Errorf("Expected 1 item in podList, got %d", len(podList))
+	}
+}
+
+func TestPods(t *testing.T) {
+
+	v := newVictimBase()
+	pod1 := newPod("app1", v1.PodRunning)
+	pod2 := newPod("app2", v1.PodPending)
+
+	client := fake.NewSimpleClientset(&pod1, &pod2)
+
+	podList, _ := v.Pods(client)
+
+	if len(podList) != 2 {
+		t.Errorf("Expected 2 item in podList, got %d", len(podList))
+	}
+}
+
+func TestDeletePod(t *testing.T) {
+
+	v := newVictimBase()
+	pod := newPod("app", v1.PodRunning)
+
+	client := fake.NewSimpleClientset(&pod)
+
+	if e := v.DeletePod(client, "app"); e != nil {
+		t.Errorf("Unexpected error %s while deleting pod", e)
+	}
+
+	if podList := getPodList(client); len(podList.Items) != 0 {
+		t.Errorf("Expected 0 items in podList, got %d", len(podList.Items))
+	}
+}
+
+func TestDeleteRandomPods(t *testing.T) {
+
+	v := newVictimBase()
+	pod1 := newPod("app1", v1.PodRunning)
+	pod2 := newPod("app2", v1.PodPending)
```

Comment:
Do you mean for `app2` to be in `Pending` state or is this just copy/paste?

Reply:
It verifies that not running pods do not try to be deleted